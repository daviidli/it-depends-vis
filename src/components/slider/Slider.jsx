import React from 'react';
import PropTypes from 'prop-types';
import { Slider as AntSlider } from 'antd';
import './Slider.scss';

const Slider = ({ topCount, setTopCount }) => (
	<AntSlider
		className="slider"
		min={1}
		max={200}
		marks={{
			1: '1',
			20: '20',
			50: '50',
			100: '100',
			200: '200'
		}}
		defaultValue={topCount}
		onAfterChange={setTopCount}
	/>
);

Slider.propTypes = {
	topCount: PropTypes.number.isRequired,
	setTopCount: PropTypes.func.isRequired
};

export default Slider;
