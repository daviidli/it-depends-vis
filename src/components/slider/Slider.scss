.slider {
	margin: 1em 2em 0 0;
}
