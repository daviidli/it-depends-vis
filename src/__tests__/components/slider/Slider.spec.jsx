import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Slider as AntSlider } from 'antd';
import { clone } from 'ramda';
import Slider from '../../../components/slider/Slider';

const defaultProps = {
	topCount: 20,
	setTopCount: jest.fn()
};

const setup = (props = defaultProps) => {
	const wrapper = shallow(<Slider {...props} />);
	return {
		wrapper,
		slider: wrapper.find(AntSlider)
	};
};

describe('Slider component', () => {
	it('should match snapshot', () => {
		const { wrapper } = setup();
		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it('should call setTopCount on after change', () => {
		const mockSetTopCount = jest.fn();
		const props = clone(defaultProps);
		props.setTopCount = mockSetTopCount;
		const { slider } = setup(props);
		expect(mockSetTopCount).toHaveBeenCalledTimes(0);
		slider.simulate('afterChange', 50);
		expect(mockSetTopCount).toHaveBeenCalledTimes(1);
		expect(mockSetTopCount).toHaveBeenCalledWith(50);
	});
});
