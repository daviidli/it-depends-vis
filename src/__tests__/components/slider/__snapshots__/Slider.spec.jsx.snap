// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider component should match snapshot 1`] = `
<Slider
  className="slider"
  defaultValue={20}
  marks={
    Object {
      "1": "1",
      "100": "100",
      "20": "20",
      "200": "200",
      "50": "50",
    }
  }
  max={200}
  min={1}
  onAfterChange={[MockFunction]}
  tipFormatter={[Function]}
/>
`;
